### PR TITLE
Improve/fix "nix-prefetch-git --fetch-lfs"

### PR DESCRIPTION
**Pull request by bjornfor**
###### Motivation for this change
`nix-prefetch-git --fetch-lfs ...` currently fails (due to unset $TMPDIR, missing git-lfs from $PATH). This fixes it.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 134122` at 7692232a run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built successfully:</summary>
  <ul>
    <li>bundix</li>
    <li>cabal2nix</li>
    <li>crate2nix</li>
    <li>dep2nix</li>
    <li>go2nix</li>
    <li>haskellPackages.cabal2nix-unstable</li>
    <li>haskellPackages.nvfetcher</li>
    <li>haskellPackages.update-nix-fetchgit</li>
    <li>nix-prefetch-git</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-update-source</li>
    <li>nvfetcher</li>
    <li>update-nix-fetchgit</li>
    <li>vgo2nix</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md">license-missing</a>

Package is missing a license.

Near pkgs/tools/package-management/nix-prefetch-scripts/default.nix:22:5:
```
   |
22 |     meta = with lib; {
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/package-management/nix-prefetch-scripts/default.nix:13:5:
```
   |
13 |     installPhase = ''
   |     ^
```
</li>
  </ul>
</details>

**Comment by bjornfor** [comment]
Merging in >=24 hours.

File: pkgs/build-support/fetchgit/nix-prefetch-git [hunk 1]
```diff
@@ -300,7 +300,9 @@ clone_user_rev() {
     local rev="${3:-HEAD}"
 
     if [ -n "$fetchLFS" ]; then
-        HOME=$TMPDIR
+        tmpHomePath="$(mktemp -d "${TMPDIR:-/tmp}/nix-prefetch-git-tmp-home-XXXXXXXX")"
```
Comment (SuperSandro2000) [inline pkgs/build-support/fetchgit/nix-prefetch-git:303]:
```suggestion
        tmpHomePath="$(mktemp -d "$TMPDIR/nix-prefetch-git-tmp-home-XXXXXXXX")"
```
TMPDIR is always populated.

File: pkgs/build-support/fetchgit/nix-prefetch-git [hunk 2]
```diff
@@ -300,7 +300,9 @@ clone_user_rev() {
...
+        exit_handlers+=(remove_tmpHomePath)
```
Comment (SuperSandro2000) [inline pkgs/build-support/fetchgit/nix-prefetch-git:304]: Can we execute the command directly?

File: pkgs/build-support/fetchgit/nix-prefetch-git (hunk 1 above)
Comment (bjornfor) [reply pkgs/build-support/fetchgit/nix-prefetch-git:303]: Hi @SuperSandro2000,

>TMPDIR is always populated.

I tested in NixOS and Ubuntu. It's not set there.

File: pkgs/build-support/fetchgit/nix-prefetch-git (hunk 2 above)
Comment (bjornfor) [reply pkgs/build-support/fetchgit/nix-prefetch-git:304]: IMHO, doing changes like that is orthogonal to this patch series. Other places use exit_handlers like that.

File: pkgs/build-support/fetchgit/nix-prefetch-git (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/build-support/fetchgit/nix-prefetch-git:303]: You are right but that reveals that yo use mktemp wrong here.

```suggestion
        tmpHomePath="$(mktemp -d nix-prefetch-git-tmp-home-XXXXXXXX)"
```
or this
```suggestion
        tmpHomePath="$(mktemp -d)"
```
is the correct syntax

File: pkgs/build-support/fetchgit/nix-prefetch-git (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/build-support/fetchgit/nix-prefetch-git:304]: What patch series? I only see 3 commits which only do this once.

Comment (bjornfor) [reply pkgs/build-support/fetchgit/nix-prefetch-git:304]: I meant the patches (commits) in this PR. The script already registers cleanup commands in the exit_handlers array, I just continued to use the existing mechanism.

File: pkgs/build-support/fetchgit/nix-prefetch-git (hunk 1 above)
Comment (bjornfor) [reply pkgs/build-support/fetchgit/nix-prefetch-git:303]: I see. This code snippet is also based on how things already are in the script. I suggest we make another PR to update the use of mktemp/TMPDIR for all occurrences.

Comment (SuperSandro2000) [reply pkgs/build-support/fetchgit/nix-prefetch-git:303]: Sure, go ahead but please don't introduce a new wrong one.

**Pull request merged**